os/include/tinyara/sched.h

```diff
@@ -86,18 +86,24 @@
 /********************************************************************************
  * Pre-processor Definitions
  ********************************************************************************/
-#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(CONFIG_ARM_MPU)
+#if defined(CONFIG_ARM_MPU) && (defined(CONFIG_APP_BINARY_SEPARATION) || defined (CONFIG_MPU_STACKGUARD))
+#define REGS_PER_REGION 3
 enum {
 	REG_RNR,
 	REG_RBAR,
 	REG_RASR
 };
 
+#if defined(CONFIG_MPU_STACKGUARD)
+#define STACKGUARD_MPU_NUM		3	/* Region number 3 is reserved for stack protection */
```

Comment:
Should this be hard-coded? If someone uses region 3 with another purpose, then what will be happened?
@kishore-sn @sangam-swami I've requested you to remove the manual allocation of region number for mpu functions at previous PR and you said you would do that. I don't accept the hard-coded mpu region at somewhere.

Regardless of above, how about moving mpu definitions at mpu.h, not in sched.h?

Reply:
Hi @sunghan-chang ,
I agree for the above request.
Lets this PR will be for stack protection?
I will be modifying the above request with new commit/PR separately. 
It will include the changes in mpu interface related APP_BINARY_SEPARATION MPU  as well as STACKGUARD.

Reply:
ok. thank you